lib/web/apps.go

```diff
@@ -196,6 +204,22 @@ func (h *Handler) createAppSession(w http.ResponseWriter, r *http.Request, p htt
 		return nil, trace.Wrap(err)
 	}
 
+	// Ensuring proxy can handle the connection is an optional step.
+	if h.preflightAppConnection != nil && req.PreflightConnection {
```

Comment:
can we trigger the preflight request before app session is created so the `DeleteAppSession` call won't be needed ?